reference/build_helpers/autotools.rst

```diff
@@ -40,69 +40,63 @@ It also works using the :ref:`environment_append <environment_append_tool>` cont
-
+You can change some variables like ``.fpic``, ``.libs``, ``.include_paths``, ``defines`` before accessing the ``vars`` to override
+an automatic value or add new values:
 
 .. code-block:: python
-   :emphasize-lines: 8, 9, 10, 11, 12, 21, 22
+   :emphasize-lines: 8, 9, 10
 
    from conans import ConanFile, AutoToolsBuildEnvironment
 
    class ExampleConan(ConanFile):
-      settings = "os", "compiler", "build_type", "arch"
-      requires = "Poco/1.7.8p3@pocoproject/stable"
-      default_options = "Poco:shared=True", "OpenSSL:shared=True"
-
-      def _run_cmd(self, command):
-        if self.settings.os == "Windows":
-            tools.run_in_windows_bash(self, command)
-        else:
-            self.run(command)
-
-      def imports(self):
-        self.copy("*.dll", dst="bin", src="bin")
-        self.copy("*.dylib*", dst="bin", src="lib")
+      ...
 
       def build(self):
          env_build = AutoToolsBuildEnvironment(self)
-         with tools.environment_append(env_build.vars):
-            self._run_cmd("./configure")
-            self._run_cmd("make")
+         env_build.fpic = True
+         env_build.libs.append("pthread")
+         env_build.defines.append("NEW_DEFINE=23")
+         env_build.configure()
+         env_build.make()
 
 
-You can change some variables like ``.fpic``, ``.libs``, ``.include_paths``, ``defines`` before accessing the ``vars`` to override
-an automatic value or add new values:
+You can use it also with ``MSYS2``/``MinGW`` subsystems installed by setting the `win_bash` parameter
+in the constructor. It will run the the ``configure`` and ``make`` commands inside a ``bash`` that
+has to be in the path or declared in ``CONAN_BASH_PATH``:
+
 
 .. code-block:: python
-   :emphasize-lines: 8, 9, 10
+   :emphasize-lines: 12
 
    from conans import ConanFile, AutoToolsBuildEnvironment
 
    class ExampleConan(ConanFile):
-      ...
+      settings = "os", "compiler", "build_type", "arch", "os_build", "arch_build"
+
+      def imports(self):
+        self.copy("*.dll", dst="bin", src="bin")
+        self.copy("*.dylib*", dst="bin", src="lib")
 
       def build(self):
-         env_build = AutoToolsBuildEnvironment(self)
-         env_build.fpic = True
-         env_build.libs.append("pthread")
-         env_build.defines.append("NEW_DEFINE=23")
+         in_win = self.os_build == "Windows"
+         env_build = AutoToolsBuildEnvironment(self, win_bash=in_win)
          env_build.configure()
          env_build.make()
 
+
 Constructor
 -----------
 
 .. code-block:: python
 
     class AutoToolsBuildEnvironment(object):
 
-        def __init__(self, conanfile)
+        def __init__(self, conanfile, win_bash=False)
 
 Parameters:
     - **conanfile** (Required): Conanfile object. Usually ``self`` in a conanfile.py
+    - **win_bash**: Defaulted to False. When True, it will run the configure/make commands inside a
```

Comment:
Follow convention for parameter description (this important for the reference section):
- **win_bash** (Optional, Defaulted to ``False``): here the explanation